tests/test_jenkins.py

```diff
@@ -311,6 +340,62 @@ def test_fetch_depth_2(self):
         self.assertRegex(req.path, '/ci/job')
         self.assertDictEqual(req.querystring, expected)
 
+    @httpretty.activate
+    def test_fetch_auth_api_token(self):
```

Comment:
I don't see anything checking that the client is really sending the authentication as is expected.